File: backend/ride-service/Sources/App/DTOs/GetRideDetailDTO.swift
import Foundation
import Models

public struct GetRideDetailDTO: Codable {
    public var id: UUID?
    public var name: String
    public var description: String?
    public var starts: Date
    public var participants: [GetParticipantDTO]
    public var latitude: Float
    public var longitude: Float
    public var participantsSum: Int
    public var seatsSum: Int
    public var passengersSum: Int
    
    public init(id: UUID? = nil, name: String, description: String? = nil, starts: Date, participants: [GetParticipantDTO], latitude: Float, longitude: Float, participantsSum: Int, seatsSum: Int, passengersSum: Int) {
        self.id = id
        self.name = name
        self.description = description
        self.starts = starts
        self.participants = participants
        self.latitude = latitude
        self.longitude = longitude
        self.participantsSum = participantsSum
        self.seatsSum = seatsSum
        self.passengersSum = passengersSum
    }
}

File: android/app/src/main/java/net/ipv64/kivop/dtos/RideServiceDTOs/GetRideDetailDTO.kt
package net.ipv64.kivop.dtos.RideServiceDTOs

import java.util.UUID
import java.time.LocalDateTime

data class GetRideDetailDTO (
    var id : UUID?,
    var name : String,
    var description : String?,
    var starts : LocalDateTime,
    var participants : List<GetParticipantDTO>,
    var latitude : Float,
    var longitude : Float,
    var participantsSum : Int,
    var seatsSum : Int,
    var passengersSum : Int,
)
